registry/remote.go

```diff
@@ -0,0 +1,64 @@
+package registry
+
+import (
+	"context"
+	"encoding/json"
+	manifest "github.com/docker/distribution/manifest/schema1"
+	dockerregistry "github.com/heroku/docker-registry-client/registry"
+	"time"
+)
+
+type Remote interface {
+	Tags(img Image) ([]string, error)
+	Manifest(img Image) (Image, error)
+	Cancel()
+}
+
+type remote struct {
+	client dockerRegistryInterface
+	cancel context.CancelFunc
+}
+
+func newRemote(client *dockerregistry.Registry, cancel context.CancelFunc) Remote {
```

Comment:
Sorry, I don't understand this. cancel is called by code using the Remote interface. It cancels any current operations. So the Tags method doesn't need to know any cancel func because it never needs to call cancel itself.